src/main/index.js

```diff
@@ -254,13 +254,18 @@ function runApp() {
 
     app.on('second-instance', (_, commandLine, __) => {
       // Someone tried to run a second instance, we should focus our window
-      if (mainWindow && typeof commandLine !== 'undefined') {
-        if (mainWindow.isMinimized()) mainWindow.restore()
-        mainWindow.focus()
-
+      if (typeof commandLine !== 'undefined') {
         const url = getLinkUrl(commandLine)
-        if (url) {
-          mainWindow.webContents.send(IpcChannels.OPEN_URL, url)
+        if (mainWindow === 'all-windows-closed') {
+          if (url) macOSnewURLWindow(url)
+          createWindow()
```

Comment:
You need to set this to true, otherwise the mainWindow variable will stay as the string.
```suggestion
          createWindow({ replaceMainWindow: true })
```

Reply:
```js
  async function createWindow(
    {
      replaceMainWindow = true,
      windowStartupUrl = null,
      showWindowNow = false,
      searchQueryText = null
    } = { }) {
```

`replaceMainWindow` is true by default

Reply:
Looks like I was focusing too much on the other issue and didn't read this bit correctly.

For the other issue I was originally going to ask you to use modern JavaScript syntax and then realised that the timeout wasn't even necessary, as you could just do the same as when the app is launched and listen for the `APP_READY` event which is ironically sent right below the code in the `OPEN_URL` listener that you modified in your previous PR.